diff --git a/node_modules/ts-node/dist/ts-internals.js b/node_modules/ts-node/dist/ts-internals.js
index 97ea748..2050fdb 100644
--- a/node_modules/ts-node/dist/ts-internals.js
+++ b/node_modules/ts-node/dist/ts-internals.js
@@ -40,6 +40,7 @@ function createTsInternalsUncached(_ts) {
         const resolved = ts.nodeModuleNameResolver(extendedConfig, combinePaths(basePath, 'tsconfig.json'), { moduleResolution: ts.ModuleResolutionKind.NodeJs }, host, 
         /*cache*/ undefined, 
         /*projectRefs*/ undefined,
+        /*conditionsOrIsConfigLookup*/ undefined,
         /*lookupConfig*/ true);
         if (resolved.resolvedModule) {
             return resolved.resolvedModule.resolvedFileName;
