From 6f0b37c46903402a1df195467990e443fd1a7f0c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Lo=C3=AFc=20Mangeonjean?= <loic@coderpad.io>
Date: Mon, 11 Mar 2024 16:44:07 +0100
Subject: [PATCH 04/58] cleanup: remove amd build to speed up editor build

---
 build/gulpfile.editor.js | 15 ++++-----------
 1 file changed, 4 insertions(+), 11 deletions(-)

diff --git a/build/gulpfile.editor.js b/build/gulpfile.editor.js
index e781d39d9e2..4c92c45941a 100644
--- a/build/gulpfile.editor.js
+++ b/build/gulpfile.editor.js
@@ -413,17 +413,10 @@ gulp.task('editor-distro',
 			util.rimraf('out-editor-min')
 		),
 		extractEditorSrcTask,
-		task.parallel(
-			task.series(
-				compileEditorAMDTask,
-				optimizeEditorAMDTask,
-				minifyEditorAMDTask
-			),
-			task.series(
-				createESMSourcesAndResourcesTask,
-				compileEditorESMTask,
-				appendJSToESMImportsTask
-			)
+		task.series(
+			createESMSourcesAndResourcesTask,
+			compileEditorESMTask,
+			appendJSToESMImportsTask
 		),
 		finalEditorResourcesTask
 	)
-- 
2.34.1

