From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Lo=C3=AFc=20Mangeonjean?= <loic@coderpad.io>
Date: Tue, 12 Mar 2024 17:03:08 +0100
Subject: [PATCH] feat: add a way to detect if localization were already used

---
 src/vs/nls.ts | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/vs/nls.ts b/src/vs/nls.ts
index 2e914738989..78ced4fe478 100644
--- a/src/vs/nls.ts
+++ b/src/vs/nls.ts
@@ -47,6 +47,12 @@ function _format(message: string, args: (string | number | boolean | undefined |
 	return result;
 }
 
+let initialized = false;
+
+export function isInitialized() {
+	return initialized;
+}
+
 /**
  * Marks a string to be localized. Returns the localized string.
  *
@@ -91,6 +97,7 @@ export function localize(data: ILocalizeInfo | string /* | number when built */,
  * depending on the target context.
  */
 function lookupMessage(index: number, fallback: string | null): string {
+	initialized = true;
 	const message = getNLSMessages()?.[index];
 	if (typeof message !== 'string') {
 		if (typeof fallback === 'string') {
