From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Lo=C3=AFc=20Mangeonjean?= <loic@coderpad.io>
Date: Mon, 11 Mar 2024 18:23:19 +0100
Subject: [PATCH] fixup! refactor: split code

---
 .../services/keybinding/browser/keybindingService.ts        | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/vs/workbench/services/keybinding/browser/keybindingService.ts b/src/vs/workbench/services/keybinding/browser/keybindingService.ts
index cbed3a6137c..956db0ac3bd 100644
--- a/src/vs/workbench/services/keybinding/browser/keybindingService.ts
+++ b/src/vs/workbench/services/keybinding/browser/keybindingService.ts
@@ -428,10 +428,14 @@ export class WorkbenchKeybindingService extends AbstractKeybindingService {
 		this._onDidUpdateKeybindings.fire();
 	}
 
+	protected getUserKeybindingItems() {
+		return this._resolveUserKeybindingItems(this.userKeybindings.keybindings, false);
+	}
+
 	protected _getResolver(): KeybindingResolver {
 		if (!this._cachedResolver) {
 			const defaults = this._resolveKeybindingItems(KeybindingsRegistry.getDefaultKeybindings(), true);
-			const overrides = this._resolveUserKeybindingItems(this.userKeybindings.keybindings, false);
+			const overrides = this.getUserKeybindingItems();
 			this._cachedResolver = new KeybindingResolver(defaults, overrides, (str) => this._log(str));
 		}
 		return this._cachedResolver;
-- 
2.34.1

