From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Lo=C3=AFc=20Mangeonjean?= <loic@coderpad.io>
Date: Fri, 9 May 2025 11:16:13 +0200
Subject: [PATCH] fix: fix editor api types

---
 build/gulpfile.editor.js | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/build/gulpfile.editor.js b/build/gulpfile.editor.js
index 4787605d068..76690996392 100644
--- a/build/gulpfile.editor.js
+++ b/build/gulpfile.editor.js
@@ -117,7 +117,16 @@ function toExternalDTS(contents) {
 		}
 
 		if (line.indexOf('declare let MonacoEnvironment') === 0) {
-			lines[i] = `declare global {\n    let MonacoEnvironment: Environment | undefined;\n}`;
+			lines[i] = [
+				'declare global {',
+				'    let MonacoEnvironment: Environment | undefined;',
+				'',
+				'    interface Window {',
+				'        MonacoEnvironment?: Environment | undefined;',
+				'    }',
+				'}',
+				''
+			].join('\n');
 		}
 
 		if (line.indexOf('\tMonacoEnvironment?') === 0) {
