From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Lo=C3=AFc=20Mangeonjean?= <loic@coderpad.io>
Date: Mon, 11 Mar 2024 16:44:07 +0100
Subject: [PATCH] cleanup: remove amd build to speed up editor build

---
 build/gulpfile.editor.js | 13 +++----------
 1 file changed, 3 insertions(+), 10 deletions(-)

diff --git a/build/gulpfile.editor.js b/build/gulpfile.editor.js
index ea10e2c1586..34576643c7d 100644
--- a/build/gulpfile.editor.js
+++ b/build/gulpfile.editor.js
@@ -365,16 +365,9 @@ gulp.task('editor-distro',
 			util.rimraf('out-editor-min')
 		),
 		extractEditorSrcTask,
-		task.parallel(
-			task.series(
-				compileEditorAMDTask,
-				bundleEditorAMDTask,
-				minifyEditorAMDTask
-			),
-			task.series(
-				createESMSourcesAndResourcesTask,
-				compileEditorESMTask,
-			)
+		task.series(
+			createESMSourcesAndResourcesTask,
+			compileEditorESMTask,
 		),
 		finalEditorResourcesTask
 	)
