From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Lo=C3=AFc=20Mangeonjean?= <loic@coderpad.io>
Date: Mon, 11 Mar 2024 17:15:40 +0100
Subject: [PATCH] cleanup: remove code already done in
 EditorFeaturesInstantiator

---
 .../editor/standalone/browser/standaloneServices.ts  | 12 ------------
 1 file changed, 12 deletions(-)

diff --git a/src/vs/editor/standalone/browser/standaloneServices.ts b/src/vs/editor/standalone/browser/standaloneServices.ts
index ed52e63fe2b..e39eb9cba91 100644
--- a/src/vs/editor/standalone/browser/standaloneServices.ts
+++ b/src/vs/editor/standalone/browser/standaloneServices.ts
@@ -88,8 +88,6 @@ import { DefaultConfiguration } from '../../../platform/configuration/common/con
 import { WorkspaceEdit } from '../../common/languages.js';
 import { AccessibilitySignal, AccessibilityModality, IAccessibilitySignalService, Sound } from '../../../platform/accessibilitySignal/browser/accessibilitySignalService.js';
 import { LogService } from '../../../platform/log/common/logService.js';
-import { getEditorFeatures } from '../../common/editorFeatures.js';
-import { onUnexpectedError } from '../../../base/common/errors.js';
 import { ExtensionKind, IEnvironmentService, IExtensionHostDebugParams } from '../../../platform/environment/common/environment.js';
 import { mainWindow } from '../../../base/browser/window.js';
 import { ResourceMap } from '../../../base/common/map.js';
@@ -1227,16 +1225,6 @@ export module StandaloneServices {
 			}
 		}
 
-		// Instantiate all editor features
-		const editorFeatures = getEditorFeatures();
-		for (const feature of editorFeatures) {
-			try {
-				instantiationService.createInstance(feature);
-			} catch (err) {
-				onUnexpectedError(err);
-			}
-		}
-
 		onDidInitialize.fire();
 
 		return instantiationService;
