From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Lo=C3=AFc=20Mangeonjean?= <loic@coderpad.io>
Date: Mon, 11 Mar 2024 18:25:28 +0100
Subject: [PATCH] fix: allow adding a local folder even when there is a remote
 connection

---
 .../workspaces/browser/abstractWorkspaceEditingService.ts  | 7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/src/vs/workbench/services/workspaces/browser/abstractWorkspaceEditingService.ts b/src/vs/workbench/services/workspaces/browser/abstractWorkspaceEditingService.ts
index be885f55a7d..6f5495a8a09 100644
--- a/src/vs/workbench/services/workspaces/browser/abstractWorkspaceEditingService.ts
+++ b/src/vs/workbench/services/workspaces/browser/abstractWorkspaceEditingService.ts
@@ -14,7 +14,7 @@ import { ConfigurationScope, IConfigurationRegistry, Extensions as Configuration
 import { Registry } from '../../../../platform/registry/common/platform.js';
 import { ICommandService } from '../../../../platform/commands/common/commands.js';
 import { distinct } from '../../../../base/common/arrays.js';
-import { basename, isEqual, isEqualAuthority, joinPath, removeTrailingPathSeparator } from '../../../../base/common/resources.js';
+import { basename, isEqual, joinPath, removeTrailingPathSeparator } from '../../../../base/common/resources.js';
 import { INotificationService, Severity } from '../../../../platform/notification/common/notification.js';
 import { IFileService } from '../../../../platform/files/common/files.js';
 import { IWorkbenchEnvironmentService } from '../../environment/common/environmentService.js';
@@ -171,11 +171,6 @@ export abstract class AbstractWorkspaceEditingService extends Disposable impleme
 
 	private async doAddFolders(foldersToAdd: IWorkspaceFolderCreationData[], index?: number, donotNotifyError: boolean = false): Promise<void> {
 		const state = this.contextService.getWorkbenchState();
-		const remoteAuthority = this.environmentService.remoteAuthority;
-		if (remoteAuthority) {
-			// https://github.com/microsoft/vscode/issues/94191
-			foldersToAdd = foldersToAdd.filter(folder => folder.uri.scheme !== Schemas.file && (folder.uri.scheme !== Schemas.vscodeRemote || isEqualAuthority(folder.uri.authority, remoteAuthority)));
-		}
 
 		// If we are in no-workspace or single-folder workspace, adding folders has to
 		// enter a workspace.
