From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Lo=C3=AFc=20Mangeonjean?= <loic@coderpad.io>
Date: Wed, 30 Oct 2024 17:47:38 +0100
Subject: [PATCH] fix: stop expecting a git context

---
 build/npm/postinstall.ts | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/build/npm/postinstall.ts b/build/npm/postinstall.ts
index 3e260853a53..e0370c4f2c7 100644
--- a/build/npm/postinstall.ts
+++ b/build/npm/postinstall.ts
@@ -182,5 +182,5 @@ for (const dir of dirs) {
 	npmInstall(dir, opts);
 }
 
-child_process.execSync('git config pull.rebase merges');
-child_process.execSync('git config blame.ignoreRevsFile .git-blame-ignore-revs');
+// child_process.execSync('git config pull.rebase merges');
+// child_process.execSync('git config blame.ignoreRevsFile .git-blame-ignore-revs');
