From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Lo=C3=AFc=20Mangeonjean?= <loic@coderpad.io>
Date: Wed, 30 Oct 2024 17:47:38 +0100
Subject: [PATCH] fix: stop expecting a git context

---
 build/npm/postinstall.ts | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/build/npm/postinstall.ts b/build/npm/postinstall.ts
index b6a934f74b3..9a20e65d782 100644
--- a/build/npm/postinstall.ts
+++ b/build/npm/postinstall.ts
@@ -216,5 +216,5 @@ for (const dir of dirs) {
 	npmInstall(dir, { env });
 }
 
-child_process.execSync('git config pull.rebase merges');
-child_process.execSync('git config blame.ignoreRevsFile .git-blame-ignore-revs');
+// child_process.execSync('git config pull.rebase merges');
+// child_process.execSync('git config blame.ignoreRevsFile .git-blame-ignore-revs');
