From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Lo=C3=AFc=20Mangeonjean?= <loic@coderpad.io>
Date: Fri, 4 Apr 2025 15:23:01 +0200
Subject: [PATCH] fix: ignore @vscode/tree-sitter-wasm until MS fixes it

---
 build/gulpfile.editor.ts | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/build/gulpfile.editor.ts b/build/gulpfile.editor.ts
index 447b76fa16c..2593f626646 100644
--- a/build/gulpfile.editor.ts
+++ b/build/gulpfile.editor.ts
@@ -55,7 +55,7 @@ const extractEditorSrcTask = task.define('extract-editor-src', () => {
 			'vs/base/common/marked/marked.js',
 		],
 		shakeLevel: 2, // 0-Files, 1-InnerFile, 2-ClassMembers
-		importIgnorePattern: /\.css$/,
+		importIgnorePattern: /\.css$|^@vscode\//,
 		destRoot: path.join(root, 'out-editor-src'),
 		tsOutDir: '../out-monaco-editor-core/esm/vs',
 	});
