From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Lo=C3=AFc=20Mangeonjean?= <loic@coderpad.io>
Date: Wed, 6 Aug 2025 15:11:40 +0200
Subject: [PATCH] fix: close auxiliary window when the context is unloaded

---
 .../auxiliaryWindow/browser/auxiliaryWindowService.ts       | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/vs/workbench/services/auxiliaryWindow/browser/auxiliaryWindowService.ts b/src/vs/workbench/services/auxiliaryWindow/browser/auxiliaryWindowService.ts
index 3adc42984d0..412c22ccf9e 100644
--- a/src/vs/workbench/services/auxiliaryWindow/browser/auxiliaryWindowService.ts
+++ b/src/vs/workbench/services/auxiliaryWindow/browser/auxiliaryWindowService.ts
@@ -154,6 +154,12 @@ export class AuxiliaryWindow extends BaseWindow implements IAuxiliaryWindow {
 			this._register(addDisposableListener(this.window.document.body, EventType.DRAG_OVER, (e: DragEvent) => EventHelper.stop(e)));	// Prevent drag feedback on <body>
 			this._register(addDisposableListener(this.window.document.body, EventType.DROP, (e: DragEvent) => EventHelper.stop(e)));		// Prevent default navigation on drop
 		}
+
+		// When the running context is unloaded
+		// eslint-disable-next-line no-restricted-globals
+		this._register(addDisposableListener(window, EventType.UNLOAD, () => {
+			this.window?.close();
+		}));
 	}
 
 	private handleBeforeUnload(e: BeforeUnloadEvent): void {
