From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Lo=C3=AFc=20Mangeonjean?= <loic@coderpad.io>
Date: Mon, 8 Dec 2025 16:12:34 +0100
Subject: [PATCH] refactor: split code to be able to override behavior

---
 .../keybinding/browser/keybindingService.ts    | 18 +++++++++++-------
 1 file changed, 11 insertions(+), 7 deletions(-)

diff --git a/src/vs/workbench/services/keybinding/browser/keybindingService.ts b/src/vs/workbench/services/keybinding/browser/keybindingService.ts
index 6929b00e4bd..61a33210678 100644
--- a/src/vs/workbench/services/keybinding/browser/keybindingService.ts
+++ b/src/vs/workbench/services/keybinding/browser/keybindingService.ts
@@ -245,17 +245,11 @@ export class WorkbenchKeybindingService extends AbstractKeybindingService {
 				return;
 			}
 
-			const keyboard: IKeyboard | null = (<INavigatorWithKeyboard>mainWindow.navigator).keyboard;
-
 			if (BrowserFeatures.keyboard === KeyboardSupport.None) {
 				return;
 			}
 
-			if (browser.isFullscreen(mainWindow)) {
-				keyboard?.lock(['Escape']);
-			} else {
-				keyboard?.unlock();
-			}
+			this.lockKeyCodes(browser.isFullscreen(mainWindow) ? ['Escape'] : []);
 
 			// update resolver which will bring back all unbound keyboard shortcuts
 			this._cachedResolver = null;
@@ -263,6 +257,16 @@ export class WorkbenchKeybindingService extends AbstractKeybindingService {
 		}));
 	}
 
+	protected lockKeyCodes(keyCodes: string[]): void {
+		const keyboard: IKeyboard | null = (<INavigatorWithKeyboard>mainWindow.navigator).keyboard;
+
+		if (keyCodes.length > 0) {
+			keyboard?.lock(keyCodes);
+		} else {
+			keyboard?.unlock();
+		}
+	}
+
 	public override dispose(): void {
 		this._contributions.forEach(c => c.listener?.dispose());
 		this._contributions.length = 0;
