From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Lo=C3=AFc=20Mangeonjean?= <loic@coderpad.io>
Date: Thu, 19 Mar 2026 12:21:16 +0100
Subject: [PATCH] feat: add AIDisabled policy

---
 .../contrib/chat/browser/chat.contribution.ts         | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/src/vs/workbench/contrib/chat/browser/chat.contribution.ts b/src/vs/workbench/contrib/chat/browser/chat.contribution.ts
index c438feda7d0..7e7c80ab65f 100644
--- a/src/vs/workbench/contrib/chat/browser/chat.contribution.ts
+++ b/src/vs/workbench/contrib/chat/browser/chat.contribution.ts
@@ -1564,6 +1564,17 @@ configurationRegistry.registerConfiguration({
 			description: nls.localize('chat.disableAIFeatures', "Disable and hide built-in AI features provided by GitHub Copilot, including chat and inline suggestions."),
 			default: false,
 			scope: ConfigurationScope.WINDOW,
+			policy: {
+				name: 'ChatAIFeaturesDisabled',
+				category: PolicyCategory.InteractiveSession,
+				minimumVersion: '1.111',
+				localization: {
+					description: {
+						key: 'chat.disableAIFeatures',
+						value: nls.localize('chat.disableAIFeatures', "Disable and hide built-in AI features provided by GitHub Copilot, including chat and inline suggestions.",)
+					}
+				}
+			}
 		},
 		[ChatConfiguration.TitleBarSignInEnabled]: {
 			type: 'boolean',
